### PR TITLE
refactor: quiz template config

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Refactors `questionBank` logic to use a simplified `questionBankConfig` object, which seeds the `questionBank` object needed for the front end

## Example

Inside `const questionBankConfig: QuestionBankConfig` from `src/data/quizzes/questionBank.ts`:

```tsx
  "some-quiz-topic": [
    { totalAnswers: 4, correctAnswer: 2 },
    { totalAnswers: 2, correctAnswer: 1 },
    { totalAnswers: 4, correctAnswer: 4 },
  ],
```

This will be parsed into the following before being exported as part of `questionBank`:

```tsx
  "some-quiz-topic-1": {
    prompt: "some-quiz-topic-1-prompt",
    answers: [
      {
        id: "some-quiz-topic-1-a",
        label: "some-quiz-topic-1-a-label",
        explanation: "some-quiz-topic-1-a-explanation",
      },
      {
        id: "some-quiz-topic-1-b",
        label: "some-quiz-topic-1-b-label",
        explanation: "some-quiz-topic-1-b-explanation",
      },
      {
        id: "some-quiz-topic-1-c",
        label: "some-quiz-topic-1-c-label",
        explanation: "some-quiz-topic-1-c-explanation",
      },
      {
        id: "some-quiz-topic-1-d",
        label: "some-quiz-topic-1-d-label",
        explanation: "some-quiz-topic-1-d-explanation",
      },
    ],
    correctAnswerId: "some-quiz-topic-1-b",
  },
  "some-quiz-topic-2": {
    prompt: "some-quiz-topic-2-prompt",
    answers: [
      {
        id: "some-quiz-topic-2-a",
        label: "some-quiz-topic-2-a-label",
        explanation: "some-quiz-topic-2-a-explanation",
      },
      {
        id: "some-quiz-topic-2-b",
        label: "some-quiz-topic-2-b-label",
        explanation: "some-quiz-topic-2-b-explanation",
      },
    ],
    correctAnswerId: "some-quiz-topic-2-a",
  },
  "some-quiz-topic-3": {
    prompt: "some-quiz-topic-3-prompt",
    answers: [
      {
        id: "some-quiz-topic-3-a",
        label: "some-quiz-topic-3-a-label",
        explanation: "some-quiz-topic-3-a-explanation",
      },
      {
        id: "some-quiz-topic-3-b",
        label: "some-quiz-topic-3-b-label",
        explanation: "some-quiz-topic-3-b-explanation",
      },
      {
        id: "some-quiz-topic-3-c",
        label: "some-quiz-topic-3-c-label",
        explanation: "some-quiz-topic-3-c-explanation",
      },
      {
        id: "some-quiz-topic-3-d",
        label: "some-quiz-topic-3-d-label",
        explanation: "some-quiz-topic-3-d-explanation",
      },
    ],
    correctAnswerId: "some-quiz-topic-3-d",
  },
```

Separately, these strings must be added to the `en/learn-quizzes.json` file for each question:

- `some-quiz-topic-${questionNumberForTopic}-prompt`
- `some-quiz-topic-${questionNumberForTopic}-${answerChoiceLetter}-label` (for each answer choice)
- `some-quiz-topic-${questionNumberForTopic}-${answerChoiceLetter}-explanation` (for each answer choice)

## Related Issue
Ongoing Quiz improvement efforts

- #14148

### CODE REVIEW

File: src/data/quizzes/questionBank.ts [hunk 1]
```diff
@@ -1,1615 +1,158 @@
...
+ * - "what-is-ethereum-1-a-label"
+ * - "what-is-ethereum-1-a-explanation"
+ * - "what-is-ethereum-1-b-label"
+ * - "what-is-ethereum-1-b-explanation"
+ * - "what-is-ethereum-1-c-label"
+ * - "what-is-ethereum-1-c-explanation"
+ * - "what-is-ethereum-1-d-label"
+ * - "what-is-ethereum-1-d-explanation"
+ */
+const questionBankConfig: QuestionBankConfig = {
```
Comment (wackerow) [inline src/data/quizzes/questionBank.ts:34]: 🚨 Start here for the core of proposed changes

**Review by corwintines — APPROVED** [review]

File: src/data/quizzes/questionBank.ts [hunk 2]
```diff
@@ -1,1746 +1,165 @@
...
-    answers: [
-      {
-        id: "run-a-node-6-a",
-        label: "run-a-node-6-a-label",
-        explanation: "run-a-node-6-a-explanation",
-      },
-      {
-        id: "run-a-node-6-b",
-        label: "run-a-node-6-b-label",
-        explanation: "run-a-node-6-a-explanation",
```
Comment (minimalsm) [inline src/data/quizzes/questionBank.ts:1477]: Like here

File: src/data/quizzes/questionBank.ts [hunk 3]
```diff
@@ -1,1746 +1,165 @@
...
+    { totalAnswers: 4, correctAnswer: 1 },
+    { totalAnswers: 4, correctAnswer: 4 },
+    { totalAnswers: 4, correctAnswer: 1 },
+  ],
+}
+
+const charFromIdx = (idx: number) => String.fromCharCode(97 + idx)
+
+// Reduce to satisfy QuestionBank data shape for use in front end
+const questionBank = Object.entries(questionBankConfig).reduce(
```
Comment (minimalsm) [inline src/data/quizzes/questionBank.ts:137]: @corwintines @wackerow we've introduced a bug here by not allowing us to reference different explanations [like here](https://github.com/ethereum/ethereum-org-website/pull/14208/files#r1881113637)

File: src/data/quizzes/questionBank.ts [hunk 4]
```diff
@@ -1,1746 +1,165 @@
...
-      },
-      {
-        id: "staking-6-b",
-        label: "staking-6-b-label",
-        explanation: "staking-6-b-explanation",
-      },
-      {
-        id: "staking-6-c",
-        label: "staking-6-c-label",
-        explanation: "staking-6-b-explanation",
```
Comment (minimalsm) [inline src/data/quizzes/questionBank.ts:1168]: And here

Screenshot 2024-12-11 at 22 45 58

File: src/data/quizzes/questionBank.ts (hunk 3 above)
Comment (wackerow) [reply src/data/quizzes/questionBank.ts:137]: I have some thoughts on a way to easily override and re-use explanations, will patch

Comment (pettinarip) [reply src/data/quizzes/questionBank.ts:137]: @wackerow could be also a idea to add a test for this case in the quiz stories.

**Pull request merged**